File: Frontend Url_Shortener(Bobby)/Url-Shortener/src/app/admin/service/transaction.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';

@Injectable({
  providedIn: 'root'
})
export class TransactionService {

  getAllTransactionApi: string = 'http://localhost:8080/urlapp/transaction/getallTransaction'

  constructor(private http: HttpClient) { }

  viewAllTransaction(pageNumber: number, pageSize: number): Observable<any> {
    const token = localStorage.getItem('token');

    const headers = {
      Authorization: `Bearer ${token}`
    };
    const params = {
      pageNumber,
      pageSize
    }
    return this.http.get<any[]>(this.getAllTransactionApi, { params })
  }


}

File: src/main/java/com/aurionpro/service/transaction/TransactionService.java
package com.aurionpro.service.transaction;

import com.aurionpro.dto.PageResponse.PageResponseDto;
import com.aurionpro.dto.transaction.TransactionResponseDto;


public interface TransactionService {
	
	PageResponseDto<TransactionResponseDto> getAllTransaction(int pagenumber,int pagesize);

	long getTotalTransaction();

}
